appinventor/components/src/com/google/appinventor/components/runtime/util/FileUtil.java

```diff
@@ -585,4 +820,261 @@ public int getErrorMessageNumber() {
+        }
+        break;
+      default:
+        break;
+    }
+    throw new IOException("Unsupported file scope: " + file.getScope());
+  }
+
+  /**
+   * Open a {@link ScopedFile} for writing as an {@link OutputStream}. The caller is responsible
+   * for closing the stream returned by this method.
+   *
+   * @param form the form to use as an Android context
+   * @param file the file to open for reading
+   * @return a new input stream.
+   * @throws IOException if the file cannot be opened
+   */
+  @SuppressWarnings("deprecation")
+  public static OutputStream openForWriting(Form form, ScopedFile file) throws IOException {
+    switch (file.getScope()) {
+      case Asset:
+        throw new IOException("Assets are read-only.");
+      case App:
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.FROYO) {
+          return new FileOutputStream(new File(Environment.getExternalStorageDirectory(),
+              file.getFileName()));
+        }
+        return new FileOutputStream(new File(form.getExternalFilesDir(""), file.getFileName()));
+      case Cache:
+        return new FileOutputStream(new File(URI.create(form.getCachePath(file.getFileName()))));
+      case Legacy:
+        return new FileOutputStream(new File(Environment.getExternalStorageDirectory(),
+            file.getFileName()));
+      case Private:
+        return new FileOutputStream(new File(URI.create(form.getPrivatePath(file.getFileName()))));
+      case Shared:
+        String[] parts = file.getFileName().split("/", 2);
+        final ContentValues values = new ContentValues();
+        values.put(MediaStore.MediaColumns.DISPLAY_NAME, parts[1]);
+        values.put(MediaStore.MediaColumns.MIME_TYPE, "");
+        values.put(MediaStore.MediaColumns.RELATIVE_PATH, parts[0]);
+        final ContentResolver resolver = form.getContentResolver();
+        Uri contentUri = getContentUriForPath(file.getFileName());
+        Uri uri = resolver.insert(contentUri, values);
+
+        if (uri == null) {
+          throw new IOException("Unable to insert MediaStore entry for shared content");
+        }
+
+        OutputStream out = resolver.openOutputStream(uri);
+        if (out == null) {
+          throw new IOException("Unable to open stream for writing");
+        }
+        return out;
+      default:
+        break;
+    }
+    throw new IOException("Unsupported file scope: " + file.getScope());
+  }
+
+  private static Uri getContentUriForPath(String path) {
+    if ("DCIM".equals(path) || "Pictures".equals(path) || "Screenshots".equals(path)) {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        return MediaStore.Images.Media.getContentUri(MediaStore.VOLUME_EXTERNAL);
+      }
+      return MediaStore.Images.Media.EXTERNAL_CONTENT_URI;
+    } else if ("Videos".equals(path) || "Movies".equals(path)) {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        return MediaStore.Images.Media.getContentUri(MediaStore.VOLUME_EXTERNAL);
```

Comment:
This should return `MediaStore.Video`, shouldn't it?

Reply:
Yes.